### PR TITLE
Name MapFlashingFlags and split up guest/staff loop

### PR DESCRIPTION
**Pull request by duncanspumpkin**

### CODE REVIEW

**Comment by IntelOrca** [comment]
Why does `DrawMapPeepPixel` need a callback to get the colour when the functions that get passed looked like they can be constants passed in in-advance?

**Comment by duncanspumpkin** [comment]
That's a good point. Not sure what I was thinking. That would be much simpler

File: src/openrct2-ui/windows/Map.cpp [hunk 1]
```diff
@@ -34,15 +34,21 @@
 #include <vector>
 
 #define MAP_COLOUR_2(colourA, colourB) (((colourA) << 8) | (colourB))
-#define MAP_COLOUR(colour) MAP_COLOUR_2(colour, colour)
-#define MAP_COLOUR_UNOWNED(colour) (PALETTE_INDEX_10 | ((colour)&0xFF00))
+#define MAP_COLOUR(flashColour) MAP_COLOUR_2(flashColour, flashColour)
+#define MAP_COLOUR_UNOWNED(flashColour) (PALETTE_INDEX_10 | ((flashColour)&0xFF00))
```
Comment (ZehMatt) [inline src/openrct2-ui/windows/Map.cpp:38]: Those are good candidates to be converted to constexpr functions I think.

Comment (duncanspumpkin) [reply src/openrct2-ui/windows/Map.cpp:38]: urgh didn't mean to rename these

File: src/openrct2-ui/windows/Map.cpp [hunk 2]
```diff
@@ -33,16 +33,31 @@
...
+    return (colourA << 8) | colourB;
+}
+constexpr uint16_t MapColour(uint8_t colour)
+{
+    return MapColour2(colour, colour);
+}
+constexpr uint16_t MapColourUnowned(uint16_t colour)
+{
+    return MapColour2((colour & 0xFF00) >> 8, PALETTE_INDEX_10);
+}
```
Comment (ZehMatt) [inline src/openrct2-ui/windows/Map.cpp:47]: Missing ```static```

**Pull request merged**